[Gui Ming Jiang] iP

mjgui opened the pull request:
# Duke

> "Wow look I'm using quotes on GitHub!" - somebody, 2k22

My PR is pretty cool because
- I
- Can
- Use
- Many
- Bullet
- Points!

How to use this application
1. Download the source code
2. Build it yourself
3. ???
4. Profit!

**WOW!** So quick and easy

## Things I have done
- [x] Fall behind schedule
- [ ] Not fall behind schedule

My main method is pretty cool too:
```java
public static void main(String[] args) {
    new Duke("duke.dat").run();
}
```

Commit:
Add Gradle support

mjgui committed:
Add greet, echo, exit behaviour

mjgui committed:
Add add and list behaviour

mjgui committed:
Add mark behaviour

mjgui committed:
Implement ToDo, Event, Deadline

mjgui committed:
Throw and handle errors

mjgui committed:
Add delete behaviour

mjgui committed:
Implement Data Save

mjgui committed:
Implement Date Parsing

mjgui committed:
Merge branch 'branch-Level-7'

mjgui committed:
Merge branch 'branch-Level-8'

mjgui committed:
Merge branch 'add-gradle-support'

mjgui committed:
Refactor with more OOP and packages

mjgui committed:
Add some junit tests

Eugene-Ong-W-X commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,44 @@
+import commands.Command;
```
Perhaps Duke.java should also be in a package?

Eugene-Ong-W-X commented on src/main/java/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,30 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        Task task = tasks.add(new Deadline(title, false, dateBy));
+        ui.printMultiMsg(new String[]{
```
```suggestion
        ui.printMultiMsg(new String[] {
```

Eugene-Ong-W-X commented on src/main/java/data/Task.java:
```diff
@@ -0,0 +1,26 @@
+
+public abstract class Task implements Serializable {
+    private final String title;
+    private boolean done;
```
Perhaps naming it isDone might make the name clearer?

Eugene-Ong-W-X commented on src/main/java/data/TaskList.java:
```diff
@@ -0,0 +1,48 @@
+        return task;
+    }
+
+    public int size() {
```
Using verbs instead of nouns for method names might be more suitable

Eugene-Ong-W-X commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,73 @@
+
+        String[] words = input.split(" ", 2);
+        command = words[0];
+        if (words.length > 1) description = words[1];
```
Body of conditionals should still be on a separate line even for a single line body

Eugene-Ong-W-X commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,73 @@
+        else description = "";
+
+        switch (command) {
+            case "bye":
```
case lines should not be indented (i.e. same level of indentation as the switch statement)

Eugene-Ong-W-X reviewed (commented):
Some minor violations of the coding standard, but no major issues. Good job!

mjgui committed:
Fix code style

mjgui committed:
Add manifest file for jar file

mjgui committed:
Add JavaDoc comments

mjgui committed:
Fix code styling

mjgui committed:
Make further code styling changes

mjgui committed:
Implement find method

mjgui committed:
Merge branch 'branch-A-JavaDoc'

mjgui committed:
Add spaces to array initializers

mjgui committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/ui/Ui.java

mjgui committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/data/TaskList.java
#	src/main/java/parser/Parser.java

seox123 commented on src/main/java/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,30 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        Task task = tasks.add(new Deadline(title, false, dateBy));
+        ui.printMultiMsg(new String[] {
+                "Got it. I've added this task:",
+                "  " + task,
+                "Now you have " + tasks.getSize() + " task" + (tasks.getSize() == 1 ? "" : "s") + " in the list."
+        });
```
I noticed that you used this same call in all 3 of the task commands. Perhaps you could consider abstracting this into a function within ui?

seox123 commented on src/main/java/commands/ToDoCommand.java:
```diff
@@ -0,0 +1,30 @@
+    private final String description;
+
+    public ToDoCommand(String description) throws DukeException {
+        if (description.length() == 0) {
+            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
```
There is a similar check before the only call of the ToDoCommand constructor. Perhaps either that check or this second check could be omitted?

seox123 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,85 @@
+package parser;
+
+import commands.ByeCommand;
+import commands.Command;
+import commands.DeadlineCommand;
+import commands.DeleteCommand;
+import commands.EventCommand;
+import commands.FindCommand;
+import commands.ListCommand;
+import commands.MarkCommand;
+import commands.ToDoCommand;
+import exceptions.DukeException;
```
I like the consistent ordering of import statements!

seox123 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,85 @@
+            description = "";
+        }
+
+        switch (command) {
```
I like the usage of switch here, making the code neat and easy to understand!

mjgui committed:
Add GUI support

mjgui committed:
Merge branch 'branch-Level-10'

mjgui committed:
Add checkstyle support

mjgui committed:
Fix checkstyle errors.

mjgui committed:
Merge branch 'branch-A-CheckStyle'

mjgui committed:
Add assertions

Assertions are used to ensure no null/empty values are being passed to
the Task constructors.

mjgui committed:
Refactor new task commands into same superclass

To reduce repetition of code, Event, Deadline and ToDo creation command
classes now inherit from a common NewTaskCommand.

mjgui committed:
Merge pull request #1 from mjgui/branch-A-Assertions

Add assertions

mjgui committed:
Merge branch 'master' into branch-A-CodeQuality

mjgui committed:
Fix styling issues

mjgui committed:
Merge pull request #2 from mjgui/branch-A-CodeQuality

A-CodeQuality

mjgui committed:
Add shorter alternative commands

mjgui committed:
Merge tag 'C-FriendlierSyntax'

mjgui committed:
Fix import order for tests

mjgui committed:
Fix bye command.

mjgui committed:
Add extra error checking for command parsing

mjgui committed:
Update javadoc header comments

mjgui committed:
Add UI screenshot

mjgui committed:
Add product title to program

mjgui committed:
Add credit to JavaFX tutorial in comments

mjgui committed:
Update product screenshot

mjgui committed:
Update Duke user guide
